### PR TITLE
add ssh client and rugged gem

### PR DESCRIPTION
**Pull request by rwaffen**
- rugged uses a native extension, so we need to install the build dependencies

### CODE REVIEW

File: puppetserver/Dockerfile [hunk 1]
```diff
@@ -110,6 +110,7 @@ RUN apt-get update && \
     apt-get clean && \
     rm -rf /var/lib/apt/lists/* && \
     gem install --no-doc r10k && \
```
Comment (bastelfreak) [inline puppetserver/Dockerfile:112]: should we add a version here to ensure we don't pull in breaking changes by accident?

Comment (rwaffen) [reply puppetserver/Dockerfile:112]: okay, will add...

File: puppetserver/Dockerfile [hunk 2]
```diff
@@ -7,7 +7,7 @@ ARG UBUNTU_CODENAME=jammy
 
 FROM ubuntu:22.04 as base
 
-ARG PACKAGES=ca-certificates\ git\ netbase\ openjdk-17-jre-headless\ ruby3.0
+ARG PACKAGES="ca-certificates git netbase openjdk-17-jre-headless ruby3.0 ruby3.0-dev openssh-client gcc make cmake pkg-config libssl-dev libc6-dev"
```
Comment (bastelfreak) [inline puppetserver/Dockerfile:10]: can we / should we uninstall the dev packages after the gems compiled their extensions?

**Review by marcusdots — APPROVED** [review]
The build is sane. It could be simplified, but that is for another PR.

**Review by bastelfreak — APPROVED** [review]

**Comment by marcusdots** [comment]
Tested successfully with Puppetfile and ssh

**Pull request merged**